src/api.js

```diff
@@ -0,0 +1,86 @@
+let debug = require('debug')('app:main');
+let APIBuilder = require('taskcluster-lib-api');
+let taskcluster = require('taskcluster-client');
+let uuid = require('uuid');
+
+let builder = new APIBuilder({
+  title:    'AMQP Messages API Documentation',
+  description: [
+    'This service is responsible for making pulse messages accessible',
+    'from browsers and cli. There are API endpoints to',
+    'bind / unbind to an exchange and pause / resume listening from a queue',
+  ].join('\n'),
+  serviceName: 'tcevents',
+  version: 'v1',
+  errorCodes: {
+    InputValidation:  400, // For JSON schema errors (bad exchange).
+  },
+  context: ['connection'],
+});
+
+builder.declare({
+  method: 'get',
+  route: '/connect/',
+  name: 'connect',
+  description: 'Connect to receive messages',
+  stability: APIBuilder.stability.experimental,
+  // Add input validation yml
+  title: 'Events-Api',
+}, async function(req, res) {
+  console.log("hello");
+  const sendEvent = (kind, data) => {
+    res.write('event: ' + kind + '\n');
+    res.write('data: ' + JSON.stringify(data) + '\n');
+    res.write('\n');
+  };
+
+  let headWritten, pingEvent;
+  try {
+
+    res.writeHead(200, {
+      Connection : 'keep-alive',
+      'Access-Control-Allow-Origin': '*',
+      'Content-Security-Policy': "connect-src 'self' http://localhost:* ;",
+      'Content-Type' : 'text/event-stream',
+      'Cache-Control' : 'no-cache',
+    });
+    headWritten = true;
+
+    // The headers are written without any errors.
+    // This means we are ready to send messages.
+    sendEvent('ready', true);
+
+    // TODO : add listener = PulseListener
+
+    const pingEvent = setInterval(() => sendEvent('ping', {time : new Date()}), 10*1000);
```

Comment:
This is not blocking.. hence, controlflow will continue till it hits `res.end();` in the _finally_ block.

For example you could use:
`await new Promise(resolve => setTimeout(resolve, 30 * 1000));`